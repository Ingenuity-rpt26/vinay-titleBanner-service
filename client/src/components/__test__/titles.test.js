import React from 'react';
import ReactDOM from 'react-dom';
import Titles from '../titles';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Titles></Titles>, div);
});


it('render titles correctly', () => {
  const {getByTestId} = render(<Titles title= {'hi'}></Titles>);
  expect(getByTestId('title-test')).toHaveTextContent('hi');
});

//test DOM cleanup
it('render titles correctly', () => {
  const {getByTestId} = render(<Titles title= {'bye'}></Titles>);
  expect(getByTestId('title-test')).toHaveTextContent('bye');
});

//test snapshot
it('matches snapshot', () => {
  const tree = renderer.create(<Titles title={'snapshot'}></Titles>).toJSON();
  expect(tree).toMatchSnapshot();
});

