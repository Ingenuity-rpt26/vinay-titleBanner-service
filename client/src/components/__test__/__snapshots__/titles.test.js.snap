// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="title-test"
>
  <div>
    snapshot
  </div>
</div>
`;
